### ISSUE TITLE
Missing label for delete revision (X) button [Accessibility ⭐ ] 

### ISSUE THREAD

**Issue by sumairq** [mozilla/perfcompare#408]
The delete revision(X) button is missing `aria-label` as a consequence it only reads `button` to the screen reader.

NVDA Speech Reader   | 
:-------------------------:|
![image](https://user-images.githubusercontent.com/60618877/226163458-c1ecc58d-aa25-431c-854f-0c37e942ff59.png)

Wave Evaluation            | 
:-------------------------:|
![image](https://user-images.githubusercontent.com/60618877/226162767-8a84306f-4aa0-4387-9bde-5523b21f7ede.png)
